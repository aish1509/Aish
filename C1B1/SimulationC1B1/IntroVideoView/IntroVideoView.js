import React from 'react';
import PropTypes from 'prop-types';
import {ContentContainer} from '@math/m180';
import {VideoPlayer, Languages, Path} from '@math/common';
import {titles} from '../../../VideoTitles';

const IntroVideoView = ({data}) => {
	const anchor_media_PATH = `${Path.Media}video/en_US/anchor/`;
	const block_number = `${data.blockIndex + 1}`;

	return (
		<ContentContainer>
			<VideoPlayer
				data-testid="VideoPlayer"
				ended
				vol={0.25}
				headerLeft={titles.C1B1}
				poster={`${anchor_media_PATH}M180_anchor_block_${block_number}_mVideos.png`}
			>
				<source
					src="https://iandevlin.github.io/mdn/video-player-with-captions/video/sintel-short.mp4"
					type="video/mp4"
				/>
				<track
					src="https://iandevlin.github.io/mdn/video-player-with-captions/subtitles/vtt/sintel-en.vtt"
					label="English subtitle"
					kind="subtitles"
					srcLang={Languages.English}
					default
				/>
			</VideoPlayer>
		</ContentContainer>
	);
};
IntroVideoView.propTypes = {
	data: PropTypes.object.isRequired
};

export default IntroVideoView;
