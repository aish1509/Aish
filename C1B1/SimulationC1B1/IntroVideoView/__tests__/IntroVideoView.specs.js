import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {act} from 'react-dom/test-utils';

import IntroVideoView from '../index';

describe('<IntroVideoVideo />', () => {
	let wrapper = null;

	beforeEach(() => {
		wrapper = shallow(
			<IntroVideoView data={{provisionedBlocks: [5], blockIndex: 7}} />
		);
	});
	it('IntroVideoView renders correctly', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
	it('Intro video loads correctly', () => {
		expect(wrapper.find('[data-testid="VideoPlayer"]').exists()).toBe(true);
	});
});
