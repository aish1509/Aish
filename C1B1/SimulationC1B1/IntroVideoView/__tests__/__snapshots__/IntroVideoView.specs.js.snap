// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroVideoVideo /> IntroVideoView renders correctly 1`] = `
<ContentContainer
  defaultStyle={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "padding": "10px",
    }
  }
>
  <WithStyles(VideoPlayer)
    data-testid="VideoPlayer"
    ended={true}
    headerLeft="Release an album"
    poster="/m180/media/video/en_US/anchor/M180_anchor_block_8_mVideos.png"
    vol={0.25}
  >
    <source
      src="https://iandevlin.github.io/mdn/video-player-with-captions/video/sintel-short.mp4"
      type="video/mp4"
    />
    <track
      default={true}
      kind="subtitles"
      label="English subtitle"
      src="https://iandevlin.github.io/mdn/video-player-with-captions/subtitles/vtt/sintel-en.vtt"
      srcLang="EN"
    />
  </WithStyles(VideoPlayer)>
</ContentContainer>
`;
